src/basic/random-util.c

```diff
@@ -123,19 +123,44 @@ void initialize_srand(void) {
         srand_called = true;
 }
 
-void random_bytes(void *p, size_t n) {
+#if RAND_MAX >= INT_MAX
+#  define RAND_STEP 4
```

Comment:
hmm, rand() returns an a signed integer, and a range between 0..RAND_MAX. That means if int is 32bit it would still only ever generate 31bit on each invocation...

I think RAND_STEP in this case should be 3